bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/EntryLocationIndex.java

```diff
@@ -93,9 +93,16 @@ public long getLocation(long ledgerId, long entryId) throws IOException {
     public long getLastEntryInLedger(long ledgerId) throws IOException {
         if (deletedLedgers.contains(ledgerId)) {
             // Ledger already deleted
-            return -1;
+            if (log.isDebugEnabled()) {
```

Comment:
I think you are changing the  semantics of this method.
As far as I could see. 
Last Entry ID -1 means ledger is deleted. 
```
for (entryId = firstEntryId; entryId < lastEntryId; entryId++) {
   // do somehting.
}
```

Since Last Entry ID is -1 it will not go in the for loop.  Throwing Bookie.NoEntryException might have undesirable effect in this case.

However, Bookie.NoEntryException is thrown when EntryID is valid but it does not exist.

Reply:
If you want to change semantics of this method you may have to change where and how this method is being used.